pkg/scheduler/algorithm/priorities/resource_bin_packing.go

```diff
@@ -0,0 +1,240 @@
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
+	schedulercache "k8s.io/kubernetes/pkg/scheduler/cache"
+)
+
+// ResourceBinPacking contains information to calculate bin packing priority.
+type ResourceBinPacking struct {
+	Resources []Resources
+}
+
+//Resources contains resource name and weight
+type Resources struct {
+	Resource v1.ResourceName
+	Weight   int
+}
+
+// NewResourceBinPacking creates a ResourceBinPackingPriorityMap.
+func NewResourceBinPacking(resources []Resources) (algorithm.PriorityMapFunction, algorithm.PriorityReduceFunction) {
+	for i, elm := range resources {
+		if elm.Weight == 0 {
+			resources[i].Weight = 1
+		}
+	}
+	resourceBinPackingPrioritizer := &ResourceBinPacking{
+		Resources: resources,
+	}
+	return resourceBinPackingPrioritizer.ResourceBinPackingPriorityMap, nil
+}
+
+// ResourceBinPackingPriorityDefault creates a resourceBinPacking based prioity
+func ResourceBinPackingPriorityDefault() *ResourceBinPacking {
+	defaultResourceBinPackingPrioritizer := &ResourceBinPacking{
+		Resources: []Resources{},
+	}
+	return defaultResourceBinPackingPrioritizer
+}
+
+// ResourceBinPackingPriorityMap is a priority function that favors nodes that have higher utlization of scare resource.
+// It will detect whether the requested scarce resource is present on a node, and then calculate a score ranging from 0 to 10
+// based total utlization (best fit)
+// - If none of the scare resource are requested, this node will be given the lowest priority.
+// - If the scarce resource is requested, the larger the resource utlization ratio, the higher the node's priority.
+func (r *ResourceBinPacking) ResourceBinPackingPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (schedulerapi.HostPriority, error) {
+	var score int
+	node := nodeInfo.Node()
+	if len(r.Resources) == 0 {
+		return schedulerapi.HostPriority{}, fmt.Errorf("resource not defined")
+	}
+	if node == nil {
+		return schedulerapi.HostPriority{}, fmt.Errorf("node not found")
+	}
+	nodeScore := float64(0)
+	weightSum := 0
+
+	for _, elm := range r.Resources {
+		nodeScore += calculateScareResourceScore(nodeInfo, pod, elm.Resource) * float64(elm.Weight)
+		weightSum += elm.Weight
+	}
+	nodeScore = nodeScore / float64(weightSum)
+	if glog.V(10) {
+		glog.Infof(
+			"%v -> %v: ResourceBinPackingPriority, Score: (%d)", pod.Name, node.Name, score,
+		)
+	}
+
+	return schedulerapi.HostPriority{
+		Host:  node.Name,
+		Score: int(math.Round(nodeScore)),
+	}, nil
+
+}
+
+// calculateScareResourceScore returns total utlization of the scare resource on the node
+func calculateScareResourceScore(nodeInfo *schedulercache.NodeInfo, pod *v1.Pod, resource v1.ResourceName) float64 {
+	allocatable := nodeInfo.AllocatableResource()
+	used := nodeInfo.RequestedResource()
+	var requested schedulercache.Resource
+	var requestedInit schedulercache.Resource
+	switch resource {
+	case v1.ResourceCPU:
```

Comment:
who nice, from the last time i reviewed, @sudeshsh  does it now use this function for both bin packing core resources like cpu, memory as well as extended resources now ?

Reply:
if it does, then we should be seeing the deletion of the old bin packing functions for core resourcen(cpu ,mem) ?

Reply:
Yes right now if we pass cpu and memory as params to bin packing it replicates the most requested resource function.

Reply:
i meant why arent we deleting the code container in most_requested.go ?

Reply:
Maybe after this PR goes in we can, maybe another PR to integrate this method which will be called by MostRequested internally.